src/contracts/Input/Value/InputPayload.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\Test\Rest\Input\Value;
+
+final class InputPayload implements \Stringable
+{
+    private string $mediaType;
+
+    private string $format;
+
+    private string $content;
+
+    private ?string $name;
+
+    public function __construct(string $mediaType, string $format, string $content, ?string $name = null)
+    {
+        $this->mediaType = $mediaType;
+        $this->format = $format;
+        $this->content = $content;
+        $this->name = $name;
+    }
+
+    public function getMediaType(): string
+    {
+        return $this->mediaType;
+    }
+
+    public function getMediaTypeWithFormat(): string
+    {
+        return "$this->mediaType+$this->format";
```

Comment:
Note that there can be cases where this will not fit. Specifically I'm thinking of `application/json` and `application/xml`, when media type is not explicitly defined (and multiple of our endpoints respect that regardless of whether we want to be strict about type or not).

Additionally, it is not clear for me (based only on this class) whether we expect prefix (`application/`) to be present as part of media type string or not?

Reply:
> Note that there can be cases where this will not fit. Specifically I'm thinking of `application/json` and `application/xml`, when media type is not explicitly defined (and multiple of our endpoints respect that regardless of whether we want to be strict about type or not).

Ibexa's input structure payload should never use generic web content types (or whatever you call it). I understand that such input can be easily processed since an endpoint should know what to expect. This however contradicts the architecture of our REST parsing dispatcher. It should be able to parse input structures independently of the context and it's the feature that makes it a powerful component.

To sum this up, I don't think that exposing Payload's direct Content-Type is a good idea, but doable. IMO Payload as a domain object should know only about its Media Type and format. Content-Type part has been delegated to the consumer (`BaseRestWebTestCase`).

**What is the actual use case for this?**

> 
> Additionally, it is not clear for me (based only on this class) whether we expect prefix (`application/`) to be present as part of media type string or not?

No, Content-Type consists of:
* vendor prefix: `application/vnd.ibexa.api.`
* Media Type
* Format, concatenated with Media Type by `+` sign.

I've revamped a bit `BaseRestWebTestCase::generateMediaTypeString` method to be able to attach format on its own as 2nd argument, so in this case maybe we don't need `getMediaTypeWithFormat`.

Fixed via: [1b5a3e1](https://github.com/ibexa/test-rest/pull/4/commits/1b5a3e1d4599f0866d7a120d93dc39df0956199c)

Reply:
To put it simply I've mistook `Content-Type` as if it was `Accept`. Consider it irrelevant :sweat_smile: